ReactNativeClient/lib/services/ResourceFetcher.js

```diff
@@ -248,6 +248,13 @@ class ResourceFetcher extends BaseService {
 		await Resource.resetStartedFetchStatus();
 		this.autoAddResources(null);
 	}
+
+	async cancelTimers() {
+		this.eventEmitter_.removeAllListeners();
+		if (this.scheduleQueueProcessIID_) clearTimeout(this.scheduleQueueProcessIID_);
```

Comment:
For good measure please set the id to null since it no longer refers to an existing timer.

Reply:
Yes, done. Also did for all the other modules that had the same.

While doing this I realised this object would not be correctly reinitialised in any subsequent instantiation of TestApp because it was a singleton. To fix, I modify the implementation of the singletons so we could also reset them.

Reply:
Done.